### PR TITLE
Fix: Use `module.this.id` Instead of `module.this.name`

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Use context id instead of name

## why
* Keep the keys unique

## references
* Could be utilized by https://github.com/cloudposse/terraform-aws-ec2-client-vpn/pull/24

### CODE REVIEW

**Comment by korenyoni** [comment]
```
TestExamplesComplete/NonCA 2022-01-27T17:53:57Z logger.go:66: /eg-ue1-test-self-signed-cert-84366.pem
    examples_complete_test.go:81: 
        	Error Trace:	examples_complete_test.go:81
        	Error:      	Not equal: 
        	            	expected: "/self-signed-cert.pem"
        	            	actual  : "/eg-ue1-test-self-signed-cert-84366.pem"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/self-signed-cert.pem
        	            	+/eg-ue1-test-self-signed-cert-84366.pem
        	Test:       	TestExamplesComplete/NonCA
TestExamplesComplete/NonCA 2022-01-27T17:53:57Z retry.go:72: terraform [output -no-color certificate_key_path]
TestExamplesComplete/NonCA 2022-01-27T17:53:57Z logger.go:66: Running command terraform with args [output -no-color certificate_key_path]
TestExamplesComplete/NonCA 2022-01-27T17:53:58Z logger.go:66: /eg-ue1-test-self-signed-cert-84366.key
    examples_complete_test.go:84: 
        	Error Trace:	examples_complete_test.go:84
        	Error:      	Not equal: 
        	            	expected: "/self-signed-cert.key"
        	            	actual  : "/eg-ue1-test-self-signed-cert-84366.key"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/self-signed-cert.key
        	            	+/eg-ue1-test-self-signed-cert-84366.key
        	Test:       	TestExamplesComplete/NonCA
```

**Review by korenyoni — DISMISSED** [review]
Looks good to me.

**Comment by korenyoni** [comment]
Added the `enhancement` label because I think this is more than a patch release.

File: test/src/examples_complete_test.go [hunk 1]
```diff
@@ -78,10 +78,10 @@ func testExamplesCompleteNonCA(t *testing.T) {
 	terraform.Apply(t, terraformOptions)
 
 	certificatePEMPath := terraform.Output(t, terraformOptions, "certificate_pem_path")
-	assert.Equal(t, "/self-signed-cert.pem", certificatePEMPath)
+	assert.Equal(t, "/eg-ue1-test-self-signed-cert" + attributes[0] + ".pem", certificatePEMPath)
```
Comment (korenyoni) [inline test/src/examples_complete_test.go:81]:
```suggestion
	assert.Equal(t, "/eg-ue1-test-self-signed-cert-" + attributes[0] + ".pem", certificatePEMPath)
```

File: test/src/examples_complete_test.go [hunk 2]
```diff
@@ -78,10 +78,10 @@ func testExamplesCompleteNonCA(t *testing.T) {
...
 
 	certificateKeyPath := terraform.Output(t, terraformOptions, "certificate_key_path")
-	assert.Equal(t, "/self-signed-cert.key", certificateKeyPath)
+	assert.Equal(t, "/eg-ue1-test-self-signed-cert" + attributes[0] + ".key", certificateKeyPath)
```
Comment (korenyoni) [inline test/src/examples_complete_test.go:84]:
```suggestion
	assert.Equal(t, "/eg-ue1-test-self-signed-cert-" + attributes[0] + ".key", certificateKeyPath)
```

File: test/src/examples_complete_test.go [hunk 3]
```diff
@@ -112,8 +112,8 @@ func testExamplesCompleteCA(t *testing.T) {
 	terraform.Apply(t, terraformOptions)
 
 	certificatePEMPath := terraform.Output(t, terraformOptions, "certificate_pem_path")
-	assert.Equal(t, "/self-signed-cert-ca.pem", certificatePEMPath)
+	assert.Equal(t, "/eg-ue1-test-self-signed-cert" + attributes[0] + ".pem", certificatePEMPath)
```
Comment (korenyoni) [inline test/src/examples_complete_test.go:115]:
```suggestion
	assert.Equal(t, "/eg-ue1-test-self-signed-cert-ca-" + attributes[0] + ".pem", certificatePEMPath)
```

File: test/src/examples_complete_test.go [hunk 4]
```diff
@@ -112,8 +112,8 @@ func testExamplesCompleteCA(t *testing.T) {
...
 
 	certificateKeyPath := terraform.Output(t, terraformOptions, "certificate_key_path")
-	assert.Equal(t, "/self-signed-cert-ca.key", certificateKeyPath)
+	assert.Equal(t, "/eg-ue1-test-self-signed-cert" + attributes[0] + ".key", certificateKeyPath)
```
Comment (korenyoni) [inline test/src/examples_complete_test.go:118]:
```suggestion
	assert.Equal(t, "/eg-ue1-test-self-signed-cert-ca-" + attributes[0] + ".key", certificateKeyPath)
```

**Review by korenyoni — CHANGES_REQUESTED** [review]

**Review by korenyoni — DISMISSED** [review]

**Review by korenyoni — APPROVED** [review]

**Review by korenyoni — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Review by aknysh — APPROVED** [review]

**Pull request merged**